api/app/postgresql.js

```diff
@@ -1,23 +1,43 @@
 const items = require("../db/items.json");
 const { Pool } = require("pg");
+const pino = require('pino');
+const logger = pino({ name: 'kuma-backend' });
+const dns = require('node:dns');
+
 
 const pool = new Pool({
   user: process.env.POSTGRES_USER || "kumademo",
   host: process.env.POSTGRES_HOST || "localhost",
   database: process.env.POSTGRES_DB || "kumademo",
   password: process.env.POSTGRES_PASSWORD || "kumademo",
   port: process.env.POSTGRES_PORT_NUM || 5432, //POSTGRES_PORT environmental variable is taken on K8S
+  idleTimeoutMillis: 10000,
+  connectionTimeoutMillis: 2000,
 });
 
 pool.on("error", (err, clients) => {
-  console.error("Unexpected error on idle client", err);
+  logger.error('error on postgresql pool', err);
   process.exit(-1);
 });
 
 const search = async (itemName) => {
-  return await pool.query(
-    `SELECT data FROM marketItems WHERE name ILIKE '%${itemName}%'`
-  );
+  logger.info('searching the DB...');
+  logger.info('pool details: ' + JSON.stringify(pool));
+  try {
+    const result = await pool.query(
+      `SELECT data FROM marketItems WHERE name ILIKE '%${itemName}%'`
+    );
+    logger.info('search result: ' + JSON.stringify(result));
+    return result;
+  } catch (err) {
+    logger.info("configured host in pool: " + pool.options.host);
+    logger.error('search error: ', err);
+    dns.lookup(pool.options.host, (err, address, family) => {
```

Comment:
Based on dns settings and if there are multiple ips behind this host I guess you could have a different dns responses and your error could be wrong. The best would be if you could get the IP from the error.